### ISSUE TITLE
GraphMask implementation differs from official

### ISSUE THREAD

**Issue by GuyAglionby** [pyg-team/pytorch_geometric#8403]
### 🐛 Describe the bug

The `Explainer` class makes sure that the model being explained is in eval mode before applying the explanation algorithm [here](https://github.com/pyg-team/pytorch_geometric/blob/ea2ab705716f02396b074cb1c6175b9224a3bf79/torch_geometric/explain/explainer.py#L203-L212). However, this departs from how the official GraphMask implementation works, which keeps the model in train mode so dropout is still enabled. They found this helped during training, see [here](https://github.com/MichSchli/GraphMask/blob/153565a28655fdabd90f3d4c4ed539437b1c4d81/code/analysers/graphmask/graphmask_analyser.py#L186).

I'm not sure what the best fix is here, if any. It might be best to keep `Explainer` as general as possible, so perhaps it'd be better to call `model.train()` in the `GraphMaskExplainer` [forward pass](https://github.com/pyg-team/pytorch_geometric/blob/ea2ab705716f02396b074cb1c6175b9224a3bf79/torch_geometric/explain/algorithm/graphmask_explainer.py#L113)?

**Comment by rusty1s** [comment]
Thanks for this interesting issue. I think we can provide this option as part of the input arguments to `GraphMaskExplainer` (with enabling of `dropout` turned on by default).

If we want to only enable `train` in `Dropout`, then this gets tricky, because dropout is either applied as a module or as a function. As such, just looping over `Dropout` modules will not necessarily work.
```python
model.train()
for m in model.modules():
    if isinstance(m, torch.nn.BatchNorm1d):
        m.track_running_stats = False
```
works (under certain conditions), but also feels a bit hacky.

**Comment by GuyAglionby** [comment]
Thanks for your feedback. I wonder if looping over `Dropout` is actually the best that we can do, as entering training mode won't activate the functional dropouts anyway. I agree that something other than the loop I proposed would be preferable!

I'm happy to send a PR on this issue. Interested to know your thoughts on the other three points, which, sorry, I edited in after initially posting.
